src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1883,6 +1879,12 @@ private Type EmitRangeExprNode(AssociativeNode node)
 
 
             var rangeExprFunc = AstFactory.BuildFunctionCall(methodName, arguments);
+
+            // We need to keep the ID of the generated fuction call stable
+            // in order to make use of caches in between different compile stages.
+            // So just copy it from the rangeExpression node
+            rangeExprFunc.InheritID(range.ID);
```

Comment:
this is the only `InheritID` used.

Reply:
So this is AST is still created twice - once in `gatherTypeInfo` and once in `emitIL` phase?

Reply:
yes

Reply:
I'm okay with this if you think that caching the AST will be too much work for little benefit.

Reply:
the cost to recreate the AST for `rangeExprFunc` seems to be very low